test/e2e/federated-service.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/kubernetes/federation/apis/federation"
+	"k8s.io/kubernetes/federation/client/clientset_generated/federation_internalclientset"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	UserAgentName = "federation-e2e-service-controller"
+	// TODO(madhusudancs): Using the same values as defined in the federated
+	// service controller. Replace it with the values from the e2e framework.
+	KubeAPIQPS   = 20.0
+	KubeAPIBurst = 30
+
+	FederatedServiceTimeout = 5 * time.Minute
+
+	FederatedServiceName = "federated-service"
+	FederatedServicePod  = "federated-service-test-pod"
+
+	// TODO: Only suppoprts IPv4 addresses. Also add a regexp for IPv6 addresses.
+	FederatedIPAddrRegexp  = `(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])`
```

Comment:
nit: This is not specific to anything federated.  Would it not be better to call it something like IPv4Regexp and put it in the utils library for other people to also use?